### PR TITLE
Forward port rel/6.3 to rel/6.4

### PR DESCRIPTION
**Pull request by nicolas2bert**
Forward port rel/6.3 to rel/6.4

### CODE REVIEW

**Comment by rahulreddy** [comment]
End to end http://ci.ironmann.io/gh/scality/Integration/9294

File: lib/api/multipartDelete.js [hunk 1]
```diff
@@ -61,8 +62,12 @@ function multipartDelete(authInfo, request, log, callback) {
...
             services.metadataValidateMultipart(metadataValParams,
                 (err, mpuBucket, mpuOverviewArray) => {
-                    if (err) {
-                        return next(err, destBucket);
+                    if (err === errors.NoSuchUpload) {
+                        log.trace('did not find valid mpu with uploadId' +
+                        `${uploadId}`, { method: 'multipartDelete' });
+                        // TODO in rel/6.3 and later: return 404 if
+                        // usEastBehavior is enabled in config
+                        return callback(null, destBucket);
```
Comment (electrachong) [inline lib/api/multipartDelete.js]: yeah sorry this is wrong, we want to collect corsHeaders first and return `callback(null, corsHeaders)`

**Comment by rahulreddy** [comment]
http://ci.ironmann.io/gh/scality/Integration/9295

**Comment by rahulreddy** [comment]
Since merge commits disappeared because of rebase, I redid and added them back.
End to end http://ci.ironmann.io/gh/scality/Integration/9297

**Comment by rahulreddy** [comment]
End to end http://ci.ironmann.io/gh/scality/Integration/9298

**Comment by rahulreddy** [comment]
End to end http://ci.ironmann.io/gh/scality/Integration/9303

**Pull request merged**